src/equal.c

```diff
@@ -418,6 +421,162 @@ bool equal_names(SEXP x, SEXP y) {
 
 // -----------------------------------------------------------------------------
 
+static struct vctrs_df_rowwise_info vec_equal_col(SEXP x,
+                                                  SEXP y,
+                                                  bool na_equal,
+                                                  struct vctrs_df_rowwise_info info,
+                                                  R_len_t n_row);
+
+static struct vctrs_df_rowwise_info df_equal_impl(SEXP x,
+                                                  SEXP y,
+                                                  bool na_equal,
+                                                  struct vctrs_df_rowwise_info info,
+                                                  R_len_t n_row);
+
+static struct vctrs_df_rowwise_info init_rowwise_equal_info(R_len_t n_row) {
+  struct vctrs_df_rowwise_info info;
+
+  // Initialize to "equality" value
+  // and only change if we learn that it differs
+  info.out = PROTECT(Rf_allocVector(LGLSXP, n_row));
+  int* p_out = LOGICAL(info.out);
+
+  for (R_len_t i = 0; i < n_row; ++i, ++p_out) {
+    *p_out = 1;
```

Comment:
I know I wrote code like this in the past, but I think in such simple loops we should use for readability:

```suggestion
    p_out[i] = 1;
```

But here the more structured `memset()` seems better so that the reader does not need to inspect carefully the loop.

Reply:
How would you memset() this?

Reply:
`memset(p_out, 1, sizeof(*p_out) * n_row)`? Or am I missing something?

Reply:
I tried that first and then had to learn how memset worked 😬 https://stackoverflow.com/questions/20951369/how-to-use-memset-for-initializing-bufffers-with-values-other-than-0

Reply:
oh yeah that makes sense.